pkg/builder3/openapi.go

```diff
@@ -315,12 +315,12 @@ func (o *openAPI) buildOpenAPISpec(webServices []common.RouteContainer) error {
 // BuildOpenAPISpec builds OpenAPI v3 spec given a list of route containers and common.Config to customize it.
 //
 // Deprecated: BuildOpenAPISpecFromRoutes should be used instead.
-func BuildOpenAPISpec(webServices []*restful.WebService, config *common.Config) (*spec3.OpenAPI, error) {
+func BuildOpenAPISpec(webServices []*restful.WebService, config *common.OpenAPIV3Config) (*spec3.OpenAPI, error) {
```

Comment:
how breaking is this? We have to wonder how much we break now when we make these changes. I'm assuming that only the apiserver is using that code though.

Reply:
cc @ryantxu since this will affect your workflow.

My understanding is that only apiserver uses this code, and the v3 path is less used than v2. It's also technically "wrong" to use the v2 config for v3 and I don't want to encourage people going down/sticking to that path. Things like `PostProcessSpec(*spec.Swagger) ...` (https://github.com/kubernetes/kube-openapi/blob/master/pkg/common/common.go#L114-L115) that takes a v2 spec is ignored when building v3.

I'm happy to create a new function if we want though.

Reply:
FWIW, the standard codegen tools do not produce code that calls this directly.  The breaking change will be in https://github.com/kubernetes/kubernetes/pull/121010 😬 
<img width="1274" alt="image" src="https://github.com/kubernetes/kube-openapi/assets/705951/24e07a6c-4b25-4d5b-ba57-3f373043e440">

But honestly it feels more like a bug fix since the v2 options do not cleanly apply to v3